src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -0,0 +1,146 @@
+
+import eu.chainfire.libsuperuser.Shell;
+
+/**
+ * Run by the syncthing service to convert syncthing events into local broadcasts.
+ * It uses SyncthingService.GetEvents to read the pending events and wait for new events.
+ */
+public class EventProcessor implements SyncthingService.OnWebGuiAvailableListener {
+
+    private static final String TAG = "EventProcessor";
+    private static final String PREF_LASTSYNCID = "last_sync_id";
+
+    public static final String EVENT_BASE_URL = "com.nutomic.syncthingandroid.event";
+
+    /**
+     * Minimum interval in ms at which the events are polled from syncthing and processed.
+     * This intervall will not wake up the device to save battery power.
+     */
+    public static final int EVENT_UPDATE_INTERVAL = 60000;
+
+    private final ScheduledExecutorService mScheduledExecutor =  Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> mEventPollerFuture;
+
+    private final Context mContext;
+    private final RestApi mApi;
+    private final LocalBroadcastManager mLocalBM;
+    private final Map<String, String> mFolderToPath = new HashMap<String, String>();
+
+    public EventProcessor(Context context, RestApi api) {
+        mContext = context;
+        mApi = api;
+        mLocalBM = LocalBroadcastManager.getInstance(mContext);
+    }
+
+    private void updateFolderMap()
+    {
+        synchronized(mFolderToPath) {
+            mFolderToPath.clear();
+            for (RestApi.Folder folder: mApi.getFolders()) {
+                mFolderToPath.put(folder.id, folder.path);
+            }
+        }
+    }
+
+    @Override
+    public void onWebGuiAvailable() {
+        updateFolderMap();
+        mEventPollerFuture = mScheduledExecutor.scheduleAtFixedRate(new Runnable() {
+
+            //-- Read the last SyncID and initialize mLastEventId
+            private long mLastEventId = 0;
+
+            @Override
+            public void run() {
+                // Restore the last event id if the event processor may have been restartet.
+                if (mLastEventId == 0) {
+                    mLastEventId = PreferenceManager.getDefaultSharedPreferences(mContext).getLong(PREF_LASTSYNCID, 0);
+                }
+
+
+                // First check if the event number ran backwards. If that's the case we've to start at zero because syncthing was restartet.
+                mApi.getEvents(0, 1, new RestApi.OnReceiveEventListener() {
+                    @Override
+                    public void onEvent(long id, String eventType, Bundle eventData) {
+                    }
+
+                    @Override
+                    public void onDone(long lastId) {
+                        // Now we know the last event ID. We check if it ran backwards and then start loading events.
+                        if (lastId < mLastEventId) mLastEventId = 0;
+
+                        if (Log.isLoggable(TAG, Log.DEBUG)) Log.d(TAG, "Reading events starting with id " + mLastEventId);
```

Comment:
Google uses this version in their code. They mandate for it as making debugging easier (less cluttered log) and being more battery friendly (less calls to the logging service).But I can't find the source of this citation currently. I'll remove it to be consistent with the rest of the code.